les/fetcher.go

```diff
@@ -410,14 +410,15 @@ func (f *lightFetcher) nextRequest() (*distReq, uint64) {
 		bestHash   common.Hash
 		bestAmount uint64
 	)
+	bestAmount--
 	bestTd := f.maxConfirmedTd
 	bestSyncing := false
 
 	for p, fp := range f.peers {
 		for hash, n := range fp.nodeByHash {
 			if !f.checkKnownNode(p, n) && !n.requested && (bestTd == nil || n.td.Cmp(bestTd) >= 0) {
 				amount := f.requestAmount(p, n)
-				if bestTd == nil || n.td.Cmp(bestTd) > 0 || amount < bestAmount {
+				if amount < bestAmount {
```

Comment:
Please take care in the first if statement the condition is `bestTd == nil || n.td.Cmp(bestTd) >= 0` while the second one is `bestTd == nil || n.td.Cmp(bestTd) > 0`.

Reply:
We should leave the `bestAmount` use 0 as the initial value. Because it doesn't make sense to make a fetch request if the total difficulty of the remote peer is same as the local node

Reply:
Thanks for pointing out. If we define the protocol that local node doesn't fetch same td then it would be more straightforward to write
```go
if !f.checkKnownNode(p, n) && !n.requested && (bestTd == nil || n.td.Cmp(bestTd) > 0) {
	amount := f.requestAmount(p, n)
	if amount < bestAmount {
		...
	}
	...
}
```
However, I think this is not true. The first if statement checked the block tree node is not yet known (not downloaded), and hence we need to send a fetch request though we know the total difficulty is the same. 
The only difference for this change is that, in the original code, it would not request a fetch if there is not any remote peer having td strictly greater than local node. And it would also be very likely that local node makes a fetch request for the same td due to smaller request amount.

Reply:
Thanks for your remark, I updated and fixed both the issues, please have look.